### ISSUE TITLE
Apply-PnPTenantTemplate : A task was canceled. At line:1 char:1

### ISSUE THREAD

**Issue by iBrooks** [SharePoint/sp-starter-kit#326]
#### Category
- [ ] Question
- [x] Bug
- [ ] Enhancement

#### Expected or Desired Behavior
For initial setup running the `Apply-PnPTenantTemplate -Path starterkit.pnp -Parameters @{"PORTALURL"="/sites/demoportal"; "MARKETINGALIAS"="demomarketing"; "HRALIAS"="demohr" }` should install the Kit.

#### Observed Behavior

The command cancels itself and returns the following error:

`Apply-PnPTenantTemplate : A task was canceled. `
`At line:1 char:1`
`+ Apply-PnPTenantTemplate -Path starterkit.pnp -Parameters @{"PORTALURL ...`
`+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~`
    `+ CategoryInfo          : WriteError: (:) [Apply-PnPTenantTemplate], TaskCanceledException`
    `+FullyQualifiedErrorId:EXCEPTION,SharePointPnP.PowerShell.Commands.Provisioning.Tenant.ApplyTenantTemplate`

#### Steps to Reproduce

Follow the 'Getting started' instructions.

Run the given command.

**Comment by eoverfield** [comment]
If you set the traceLog, what additional information is provided? 

You may want to review the section on the common provisioning results page regarding general cascading errors:
https://github.com/SharePoint/sp-starter-kit/blob/master/documentation/common-provision-results.md

**Comment by iBrooks** [comment]
Yes. I did look at the document that is recommended to look at if you have issues. Thank you for linking it again. If you have a look at that document, you'll see that the error I have reported here is not among the errors that the document helps with. I have also gone through all the steps in the 'General Cascading Errors' section. 

I did not find the trace to elucidate the issue at all so I had not originally included it:

`  powershell.exe Information: 0 : 2019-11-08 11:45:36.0551        [OfficeDevPnP.Core]     [0]     [Information]   File starterkit.pnp retrieved from folder       0ms                                                                             powershell.exe Information: 0 : 2019-11-08 11:45:36.1645        [OfficeDevPnP.Core]     [0]     [Information]   File starterkit.pnp retrieved from folder       0ms                                                                             powershell.exe Information: 0 : 2019-11-08 11:45:36.7270        [OfficeDevPnP.Core]     [0]     [Information]   File starterkit.xml retrieved from folder       0ms                                                                             powershell.exe Information: 0 : 2019-11-08 11:45:36.9614        [Provisioning]  [9]     [Debug] Code execution scope started    0ms     ae03ad25-288a-4c1f-a744-37c26dd0930a                                                                    powershell.exe Information: 0 : 2019-11-08 11:45:36.9614        [Provisioning]  [9]     [Information]   ProgressDelegate registered     0ms     ae03ad25-288a-4c1f-a744-37c26dd0930a                                                            powershell.exe Information: 0 : 2019-11-08 11:45:36.9614        [Provisioning]  [9]     [Information]   MessagesDelegate registered     1ms     ae03ad25-288a-4c1f-a744-37c26dd0930a                                                            powershell.exe Information: 0 : 2019-11-08 11:45:36.9770        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-core.en-us.resx retrieved from folder  0ms
powershell.exe Information: 0 : 2019-11-08 11:45:36.9927        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-core.es-es.resx retrieved from folder  0ms
powershell.exe Information: 0 : 2019-11-08 11:45:36.9927        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-core.fr-fr.resx retrieved from folder  0ms
powershell.exe Information: 0 : 2019-11-08 11:45:36.9927        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-core.de-de.resx retrieved from folder  0ms
powershell.exe Information: 0 : 2019-11-08 11:45:36.9927        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-core.nl-nl.resx retrieved from folder  0ms
powershell.exe Information: 0 : 2019-11-08 11:45:36.9927        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-core.nb-no.resx retrieved from folder  0ms
powershell.exe Information: 0 : 2019-11-08 11:45:36.9927        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-core.sv-se.resx retrieved from folder  0ms
powershell.exe Information: 0 : 2019-11-08 11:45:36.9927        [OfficeDevPnP.Core]     [0]     [Information]   File resources\resources-core.tr-tr.resx retrieved from folder  0ms
powershell.exe Information: 0 : 2019-11-08 11:45:36.9927        [ProvisioningTemplate WebHook Call]     [9]     [Debug]
Code execution scope started    0ms     ae03ad25-288a-4c1f-a744-37c26dd0930a
powershell.exe Information: 0 : 2019-11-08 11:45:36.9927        [ProvisioningTemplate WebHook Call]     [9]     [Debug]
Code execution scope ended      0ms     ae03ad25-288a-4c1f-a744-37c26dd0930a
powershell.exe Information: 0 : 2019-11-08 11:45:37.0083        [Tenant Settings]       [9]     [Debug] Code execution scope started    0ms     ae03ad25-288a-4c1f-a744-37c26dd0930a
powershell.exe Information: 0 : 2019-11-08 11:45:41.0385        [OfficeDevPnP.Core]     [0]     [Information]   File sharepoint-starter-kit.sppkg retrieved from folder ..\package      0ms
powershell.exe Information: 0 : 2019-11-08 11:49:22.1744        [Tenant Settings]       [9]     [Debug] Code execution scope ended      225157ms        ae03ad25-288a-4c1f-a744-37c26dd0930a
powershell.exe Information: 0 : 2019-11-08 11:49:22.1744        [ProvisioningTemplate WebHook Call]     [9]     [Debug]
Code execution scope started    0ms     ae03ad25-288a-4c1f-a744-37c26dd0930a
powershell.exe Information: 0 : 2019-11-08 11:49:22.1744        [ProvisioningTemplate WebHook Call]     [9]     [Debug]
Code execution scope ended      0ms     ae03ad25-288a-4c1f-a744-37c26dd0930a
powershell.exe Information: 0 : 2019-11-08 11:49:22.1744        [Provisioning]  [9]     [Debug] Code execution scope ended      225221ms        ae03ad25-288a-4c1f-a744-37c26dd0930a`

**Comment by iBrooks** [comment]
Following up, the issue has been resolved and I was able to successfully install the parts. However I do not have a specific solution. When I got back to this project after the weekend I went through and confirmed all my steps again. I also deleted the partial install from the app catalog so I could try again. When I ran the initial command again, this time it worked.

The only thing I can think of is that It had not been 24hours between setting our tenant to 'targeted release' and my first attempt at install, only ~5hrs. I'm not sure how to confirm or disprove this theory, so YMMV, but if you get this same error I did, make sure to wait the full 24 hours and dont forget to clean out the failed install from the app catalog.
